src/geometry.rs

```diff
@@ -0,0 +1,498 @@
+use nalgebra as na;
+use nalgebra::base::Vector3;
+use nalgebra::geometry::Point3;
+
+use crate::convert::cast_u32;
+
+/// Geometric data containing multiple lists of possibly _variable
+/// lengths_, such as vertices and normals, and indices - a single
+/// list containing the index topology that describes the structure of
+/// the data in those lists.
+///
+/// One such index type (`Index::TriangleFace`) binds vertices and
+/// normals in triangular faces. `Index::TriangularFace` is always
+/// ensured to have counter-clockwise winding.
+///
+/// The geometry data lives in right-handed coordinate space with the
+/// XY plance being the ground and Z axis growing upwards.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Geometry {
+    indices: Vec<Index>,
```

Comment:
I believe what is in the code essentially conforms to what you wrote. The lists of data (vertices, normals) are variable-length. They are brought together by indices, which could potentially be many types, but are just of one type at the moment: `TriangleFaceIndex`.

> You are being very specific with naming vertices and normals, so why not calling these faces? You might in the future need to implement other indexing data, such as UVW attribution etc.

So the other kinds of indices (those that are not faces) should not be in the list of indices, but rather in a separate list? I can't really imagine what other kinds of indices we will need in the future, so you know better if they should be in a single list or multiple.

In that case, yeah, we can have just `faces: Vec<Face>` where:

```rust
enum Face {
    Triangle(TriangleFace),
    Quad(QuadFace),
}

struct TriangleFace {
    pub vertices: (u32, u32, u32),
    pub normals: Option<(u32, u32, u32)>,
}

struct QuadFace {
    pub vertices: (u32, u32, u32, u32),
    pub normals: Option<(u32, u32, u32, u32)>,
}
```

Since we don't have any other kind of indices now anyway, I'll make this change to the code now.

Reply:
Done